src/rules/no-missing-label-refs.js

```diff
@@ -21,106 +21,67 @@ import { findOffsets, illegalShorthandTailPattern } from "../util.js";
 // Helpers
 //-----------------------------------------------------------------------------
 
-const labelPatterns = [
-	// [foo][bar]
-	/\]\[([^\]]+)\]/u,
-
-	// [foo][]
-	/(\]\[\])/u,
-
-	// [foo]
-	/\[([^\]]+)\]/u,
-];
-
-const shorthandTailPattern = /\]\[\]$/u;
-
 /**
  * Finds missing references in a node.
  * @param {TextNode} node The node to check.
- * @param {string} docText The text of the node.
+ * @param {string} nodeText The text of the node.
  * @returns {Array<{label:string,position:Position}>} The missing references.
  */
-function findMissingReferences(node, docText) {
+function findMissingReferences(node, nodeText) {
 	const missing = [];
-	let startIndex = 0;
-	const offset = node.position.start.offset;
 	const nodeStartLine = node.position.start.line;
 	const nodeStartColumn = node.position.start.column;
 
-	/*
-	 * This loop works by searching the string inside the node for the next
-	 * label reference. If there is, it reports an error.
-	 * It then moves the start index to the end of the label reference and
-	 * continues searching the text until the end of the text is found.
-	 */
-	while (startIndex < node.value.length) {
-		const value = node.value.slice(startIndex);
-
-		const match = labelPatterns.reduce((previous, pattern) => {
-			if (previous) {
-				return previous;
-			}
-
-			return value.match(pattern);
-		}, null);
-
-		// check for array instead of null to appease TypeScript
-		if (!Array.isArray(match)) {
-			break;
-		}
+	const labelPattern = /\[(?<left>[^\]]*)\](?:\[(?<right>[^\]]*)\])?/dgu;
+	let match;
 
+	while ((match = labelPattern.exec(nodeText))) {
```

Comment:
Can you add a comment explaining how this loop works?

Reply:
Added in https://github.com/eslint/markdown/pull/290/commits/bcee538f7bf899dae3d79dcb349afae002a8d726.